use 2.3.1 ruby with bundler cache

drh-stanford opened the pull request:
I was having problems with ruby 2.2.x, so require 2.3.x in gemspec.

eliotjordan commented:
I haven't tried running this on 2.2.x. What issues crop up? Seems like we should try to support all the rubies that the latest version of rails supports.  > 2.2.2 for rails 5.

drh-stanford commented:
It could have been my rvm setup -- I ended up nuking my entire rvm installation so I can't try 2.2.x again.

Anyway, I changed the gemspec to require `>= 2.2.2` and added that ruby version to travis.
